context/app/api/vitessce_confs/base_confs.py

```diff
@@ -428,3 +427,62 @@ def _setup_view_config_raster_cellsets_expression_segmentation(self, vc, dataset
             variablesLabelOverride="antigen", transpose=True
         )
         return vc
+
+
+class MultiImageSPRMAnndataViewConfigError(Exception):
+    """Raised when one of the individual SPRM view configs errors out"""
+    pass
+
+
+class MultiImageSPRMAnndataViewConfBuilder(ViewConfBuilder):
+    """Wrapper class for generating multiple "second generation" AnnData-backed SPRM
+    Vitessce configurations via SPRMAnnDataViewConfBuilder,
+    used for datasets with multiple regions.
+    """
+
+    def __init__(self, entity, groups_token, is_mock=False):
+        super().__init__(entity, groups_token, is_mock)
+        self._expression_id = 'expr'
+        self._mask_id = 'mask'
+        self._image_pyramid_subdir_regex = SPRM_PYRAMID_SUBDIR
+        self._mask_pyramid_subdir_regex = SPRM_PYRAMID_SUBDIR.replace(
+            self._expression_id, self._mask_id
+        )
+
+    def _find_ids(self):
+        """Search the image pyramid directory for all of the names of OME-TIFF files
+        to use as unique identifiers.
+        """
+        file_paths_found = [file["rel_path"] for file in self._entity["files"]]
+        full_pyramid_path = IMAGE_PYRAMID_DIR + "/" + self._image_pyramid_subdir_regex
+        pyramid_files = [file for file in file_paths_found if full_pyramid_path in file]
+        found_ids = [re.sub(r".ome.tif(f?)", "", Path(image_path).name).replace(
+            "_" + self._expression_id, "") for image_path in pyramid_files]
+        if len(found_ids) == 0:
+            raise FileNotFoundError(
+                f"Could not find images of the SPRM analysis with uuid {self._uuid}"
+            )
+        return found_ids
+
+    def get_conf_cells(self):
+        found_ids = self._find_ids()
+        confs = []
+        for id in sorted(found_ids):
+            vc = SPRMAnnDataViewConfBuilder(
+                entity=self._entity,
+                groups_token=self._groups_token,
+                is_mock=self._is_mock,
+                base_name=id,
+                imaging_path=self._image_pyramid_subdir_regex,
+                mask_path=self._mask_pyramid_subdir_regex,
+                image_name=f"{id}_{self._expression_id}",
+                mask_name=f"{id}_{self._mask_id}"
+            )
+            conf = vc.get_conf_cells().conf
+            if conf == {}:
+                raise MultiImageSPRMAnndataViewConfigError(
+                    f"Cytokit SPRM assay with uuid {self._uuid} has empty view\
+                        config for id '{id}'"
+                )
+            confs.append(conf)
+        return ConfCells(confs if len(confs) > 1 else confs[0], None)
```

Comment:
Could you simplify to:
```suggestion
        return ConfCells(confs, None)
```
Consistent typing of responses makes the code a lot simpler... and even if the downstream code already handles it, I think there's a good argument for moving in a direction where interfaces a simpler.

Reply:
This is a separate issue, I think.  We have been doing this for a while and the portal relies on the list/not-list to know whether to render a drop down.  Maybe a separate issue + PR?